lib/pages/chat/chat.dart

```diff
@@ -955,25 +964,35 @@ class ChatController extends State<Chat>
   }
 
   Future<void> scrollToEventId(String eventId, {bool highlight = true}) async {
-    final eventIndex = timeline!.events.indexWhere((e) => e.eventId == eventId);
+    final foundEvent =
+        timeline!.events.firstWhereOrNull((event) => event.eventId == eventId);
+
+    final eventIndex = foundEvent == null
+        ? -1
+        : timeline!.events
+            .indexWhere((event) => event.eventId == foundEvent.eventId);
```

Comment:
still not understand this, can separate to a method

Reply:
Done.